app/src/test/java/fi/helsinki/opintoni/web/rest/privateapi/portfolio/PrivateFilesResourceTest.java

```diff
@@ -18,45 +18,67 @@
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.fileUpload;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 public class PrivateFilesResourceTest extends SpringTest {
 
     private static final String CONTROL_RESOURCE_URL = "/api/private/v1/portfolio/files";
-    private static final String PUBLIC_RESOURCE_URL = "/api/public/v1/portfolio/files/student/en/olli-opiskelija/test.txt";
-    private static final String RESTRICTED_RESOURCE_URL = "/api/restricted/v1/portfolio/files/student/en/olli-opiskelija/test.txt";
-    private static final String PRIVATE_RESOURCE_URL = "/api/private/v1/portfolio/files/student/en/olli-opiskelija/test.txt";
+    private static final String PUBLIC_RESOURCE_URL = "/api/public/v1/files";
+    private static final String TEST_FILE_URL = "http://localhost//api/public/v1/files/olli-opiskelija/.*/test.txt";
     private static final String TEST_FILE_NAME = "test.txt";
     private static final String TEST_FILE_CONTENT = "test";
 
     @Test
     public void thatPortfolioFileIsSaved() throws Exception {
-        performPostFile().andExpect(status().isNoContent());
+        MvcResult result = performPostFile().andExpect(status().isOk())
+            .andExpect(jsonPath("$.uploaded").value(1))
+            .andExpect(jsonPath("$.fileName").value(TEST_FILE_NAME))
+            .andReturn();
 
-        performGetOwnedFile()
+        String url = new JSONObject(result.getResponse().getContentAsString()).get("url").toString();
+        assertTrue(url.matches(TEST_FILE_URL));
+
+        String filePath = getFilePath(url);
+
+        mockMvc.perform(get(CONTROL_RESOURCE_URL)
+            .with(securityContext(studentSecurityContext())))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$").isArray())
+            .andExpect(jsonPath("$", hasSize(1)))
+            .andExpect(jsonPath("$[0].name").value(filePath));
+
+        performGetOwnedFile(filePath)
             .andExpect(status().isOk())
             .andExpect(content().string(TEST_FILE_CONTENT));
     }
 
     @Test
     public void thatPortfolioFileIsRemoved() throws Exception {
-        performPostFile().andExpect(status().isNoContent());
-        performGetOwnedFile().andExpect(status().isOk());
+        String url = postAndGetFileUrl();
+        String filePath = getFilePath(url);
+        performGetOwnedFile(filePath).andExpect(status().isOk());
```

Comment:
Miksi tässä haetaan tiedosto public apin kautta? Lisäksi, ei kai tiedoston lataus nyt aseta tiedostoa julkiseksi by default? Ainoastaan sellaisten tiedostojen pitäisi olla julkisia jotka liittyvät käyttäjän itse julkiseksi asettamaan kenttään. 

Lisäksi voisi nimetä uudelleen `performGetOwnedFile` -> `performGetFile`/`getFile` etc

Reply:
Eikös tuon RTE:n käytön pointtina ollut, että tiedostoon voidaan tehdä linkki tekstiin? Tällöinhän sen linkin pitää pysyä samana, ei voi olla erillisiä private ja public endpointteja. Jos näkyvyys määräytyisi kentän tai portfolion perusteella, pitäisi kaikkiin kolmeen portfolioon ladata tiedosto erikseen.

Reply:
Joo, linkki pitää asettaa sellaiseksi mikä toimii kaikilla näkyvyyksillä, eli public apin kautta haku on ok.

Ongelma on nyt kuitenkin se että tiedoston saa haettua public apin kautta vaikka käyttäjä haluaisi sen näkyvän vain itselleen. Käyttäjä kuitenkin todennäköisesti odottaa tiedoston olevan sillä näkyvyydellä mikä tiedostoon linkittävällä tekstikentällä on, mikä voi olla myös privaatti.

Ideaalista olisi että tiedostoa haettaessa tarkistettaisiin onko se liitetty julkisesti näkyvään tekstikenttään ja palautetaan se vain siinä tapauksessa että on (muutoin annetaan 404). Koska tämän tarkistaminen vaatisi kuitenkin tiedostojen linkittämistä niihin liittyviin tekstikenttiin ja ko. tiedon tallentamista kantaan, voi tämä olla näin toistaiseksi. Tähän täytyy kuitenkin rakentaa tällainen myöhemmin, tai vastaavasti näyttää käyttäjälle tiedoston latauksen yhteydessä selkeä ilmoitus siitä että kaikki ladatut tiedostot tulee näkyviin julkisesti.

Lisäksi tosiaan toi apumetodin nimi on hyvä vaihtaa: `performGetOwnedFile` -> `performGetFile`/`getFile` etc